### PR TITLE
issue #7: packages with version number in json output

### PR DESCRIPTION
**Pull request by yassinjabri**
I've added the version number of the packages to the json output. The property names changed to packages as a list of package with name and version

### CODE REVIEW

**Review by aaronpowell — CHANGES_REQUESTED** [review]
Did you change the line endings? It's appearing as a full file edit, which makes me think that the line endings changed.

Please review the change so that only the lines edited as changed, not the whole file.

**Comment by yassinjabri** [comment]
I have fixed the Line Endings. Sorry that i havent recognized this.

File: src/DotNetDelice/Output.fs [hunk 1]
```diff
@@ -9,15 +9,22 @@ open Newtonsoft.Json.Serialization
 open System.IO
 open Spdx
 
+[<JsonObjectAttribute(NamingStrategyType = typeof<CamelCaseNamingStrategy>)>]
+type Package(name: string, version: string, url: string) =
```
Comment (aaronpowell) [inline src/DotNetDelice/Output.fs]: Can you please use a Record Type not a class here, there shouldn't be any need for a class.

File: src/DotNetDelice/Output.fs [hunk 2]
```diff
@@ -40,12 +47,12 @@ let private prettyPrinter printable =
...
     colorprintfn "│ └── Included deprecated IDs: $green[%b]" printable.IsDeprecatedType
     printfn "└─┬ Packages:"
-    printable.Names
+    printable.Packages
     |> Seq.iteri (fun i l ->
            let prefix =
                if i = (printable.Count - 1) then "└"
                else "├"
-           printfn "  %s── %s" prefix l)
+           printfn "  %s── %s" prefix l.Name)
```
Comment (aaronpowell) [inline src/DotNetDelice/Output.fs]: The pretty print output has now lost the URL from the output as a result of the change. I would prefer it kept that in.

**Review by aaronpowell — CHANGES_REQUESTED** [review]
It looks like this also includes the changes that are for PR #10. I don't actually mind a single PR covering both, as I see value in having both included.

File: src/DotNetDelice/Output.fs (hunk 1 above)
Comment (yassinjabri) [reply src/DotNetDelice/Output.fs]: i changed class to record type in issue #9 and issue #7. :)

File: src/DotNetDelice/Output.fs (hunk 2 above)
Comment (yassinjabri) [reply src/DotNetDelice/Output.fs]: pretty print should now be fixed  in issue #9 and issue #7

File: src/DotNetDelice/Output.fs [hunk 3]
```diff
@@ -40,12 +47,12 @@ let private prettyPrinter printable =
...
     colorprintfn "│ └── Included deprecated IDs: $green[%b]" printable.IsDeprecatedType
     printfn "└─┬ Packages:"
-    printable.Names
+    printable.Packages
     |> Seq.iteri (fun i l ->
            let prefix =
                if i = (printable.Count - 1) then "└"
                else "├"
-           printfn "  %s── %s" prefix l)
+           printfn "  %s── %s (%s)" prefix l.Name l.Url)
```
Comment (aaronpowell) [inline src/DotNetDelice/Output.fs]: This isn't an ideal output, it results in this:

```
License Expression: MIT
├── There are 3 occurances of MIT
├─┬ Conformance:
│ ├── Is OSI Approved: true
│ ├── Is FSF Free/Libre: true
│ └── Included deprecated IDs: false
└─┬ Packages:
  ├── FSharp.Core (https://licenses.nuget.org/MIT)
  ├── Newtonsoft.Json (https://raw.github.com/JamesNK/Newtonsoft.Json/master/LICENSE.md)
  └── System.ComponentModel.Annotations (https://github.com/dotnet/corefx/blob/master/LICENSE.TXT)
```

I don't want to have the URL output unless it's a package that wasn't able to determine the license. This is the reason the `name` is currently generated inline.

Comment (yassinjabri) [reply src/DotNetDelice/Output.fs]: Sorry and thank you for the hint. The output should now be fixed.

**Comment by aaronpowell** [comment]
I had to manually merge this as the branch had a merge conflict.

Also, the last change you did as part of https://github.com/aaronpowell/dotnet-delice/pull/8/commits/ebb2fcbdb1522f34c0e59a5fd0114eec05991214 didn't actually compile, nor did it really address the problem correctly. In future, please make sure that a) things compile and b) the objectives are being met.

**Pull request merged**